core/utils/aria.js

```diff
@@ -0,0 +1,470 @@
+/**
+ * @license
+ * Visual Blocks Editor
+ *
+ * Copyright 2019 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Constant declarations for common key codes.
+ * These methods are not specific to Blockly, and could be factored out into
+ * a JavaScript framework such as Closure.
+ * @author samelh@google.com (Sam El-Husseini)
+ */
+'use strict';
+
+
+goog.provide('Blockly.utils.aria');
+
+
+/**
+ * ARIA states/properties prefix.
+ * @private
+ */
+Blockly.utils.aria.ARIA_PREFIX_ = 'aria-';
+
+/**
+ * ARIA role attribute.
+ * @private
+ */
+Blockly.utils.aria.ROLE_ATTRIBUTE_ = 'role';
+
+/**
+ * ARIA role values.
+ * Copied from Closure's goog.a11y.aria.Role
+ * @enum {string}
+ */
+Blockly.utils.aria.Role = {
+  // ARIA role for an alert element that doesn't need to be explicitly closed.
+  ALERT: 'alert',
+
+  // ARIA role for an alert dialog element that takes focus and must be closed.
+  ALERTDIALOG: 'alertdialog',
```

Comment:
Although these were copied from closure, some of the constants aren't following the style guide because constants should use CONSTANT_CASE (separating words with underscore).
Since we are redefining them, I think changing the names should be fine, unless there's a reason why some, like MENU_ITEM, use constant case and others don't.

Reply:
Makes sense, I'll update.

Reply:
Actually I think the style guide is for constants, and not Enums. I think there's something to be said here about not updating them. These are meant to be 1:1 mappings to strings, just all caps, making sure the compiler catches any typos. 
If we change them, then you constantly need to think is checkbox, CHECK_BOX or CHECKBOX.

Reply:
technically guide defines the values inside an enum as constants, but 1:1 mapping makes sense and as long as we apply decision consistently (like with MENU_ITEM) I am happy with change.